src/main/java/main/AccountServiceImpl.java

```diff
@@ -0,0 +1,243 @@
+package main;
+
+import db.datasets.*;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+import org.jetbrains.annotations.NotNull;
+import rest.UserProfile;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import supportclasses.*;
+
+/**
+ * MOSch-team test server for "Kill The Birds" game
+ */
+
+public class AccountServiceImpl implements AccountService {
+    final SessionFactory sessionFactory;
+    private final Map<String, UserProfile> loggedUsers = new ConcurrentHashMap<>();
+    private final Map<Long, String> sessionIDs = new ConcurrentHashMap<>();
+
+    public AccountServiceImpl(CreationStrategy strategy) {
+        final Configuration configuration = new Configuration();
+        configuration.addAnnotatedClass(UserDataSet.class);
+        configuration.configure("hibernate.cfg.xml");
+        switch (strategy) {
+            case OPEN_OR_CREATE:
+                configuration.setProperty("hibernate.hbm2ddl.auto", "validate");
+                break;
+            case CREATE_NEW:
+                configuration.setProperty("hibernate.hbm2ddl.auto", "create");
+                break;
+            case CREATE_TEST:
+                configuration.setProperty("hibernate.connection.url", "jdbc:mysql://localhost:3306/db_java_test");
+                configuration.setProperty("hibernate.hbm2ddl.auto", "create-drop");
+                break;
+        }
+        sessionFactory = configuration.buildSessionFactory();
+    }
+
+    @SuppressWarnings("MagicNumber")
+    @NotNull
+    public static String getMD5(String input) {
+        try {
+            final MessageDigest md = MessageDigest.getInstance("MD5");
+            final byte[] messageDigest = md.digest(input.getBytes());
+            // Convert to hex string
+            final StringBuilder sb = new StringBuilder();
+            for (byte aMessageDigest : messageDigest) {
+                if ((0xff & aMessageDigest) < 0x10) {
+                    sb.append('0');
+                }
+                sb.append(Integer.toHexString(0xff & aMessageDigest));
+            }
+            return sb.toString();
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public List<UserProfile> getAllUsers() {
+        final Session session = sessionFactory.openSession();
```

Comment:
Любая работа с базой должна выполняться в блоке аналогичном следующему:

Session session;
try {
 session = sessionFactory.openSession();
    // Since we are in CMT, a JTA transaction would
    // already have been started.  This call essentially
    // no-ops
    session.getTransaction().begin();

    Number customerCount = (Number) session.createQuery( "select count(c) from Customer c" ).uniqueResult();

    // Since we did not start the transaction ( CMT ),
    // we also will not end it.  This call essentially
    // no-ops in terms of transaction handling.
    session.getTransaction().commit();
}
catch ( Exception e ) {
    LOGGER.error("Database operation failed", e)
    // again, the rollback call here would no-op (aside from
    // marking the underlying CMT transaction for rollback only).
    if ( session.getTransaction().getStatus() == TransactionStatus.ACTIVE
            || session.getTransaction().getStatus() == TransactionStatus.MARKED_ROLLBACK ) {
        session.getTransaction().rollback();
    }
    // handle the underlying error
}
finally {
    session.close();
}
http://docs.jboss.org/hibernate/orm/5.1/userguide/html_single/Hibernate_User_Guide.html#transactions-api